src/main/java/FreedomPointsCounter.java

```diff
@@ -50,7 +49,8 @@ private void checkAllFreedomLinesFrom(@NotNull Position position) {
     }
 
     private void checkFreedomLine(@NotNull Position startingPosition, @NotNull Direction direction) {
-        Stone.Color stoneColor = board.getStone(startingPosition).getColor();
+        Stone startingStone = Objects.requireNonNull(board.getStone(startingPosition), "Should be not-null, checked by count method");
```

Comment:
It's not entirely clear to me whether the other cases need to be checked for `null` values. Please take a look yourself as well

Reply:
Ok, I will refactor this thing and I will add some documentation